### ISSUE TITLE
Version3 doesn't retry

### ISSUE THREAD

**Issue by casey-juul** [nordicsemi/IOS-DFU-Library#62]
If the DFU process was interrupted due to a disconnect, Version2 would retry the process.

Version3 doesn't seem to do so. Is it now up to us to re-scan and re-initiate?

**Comment by philips77** [comment]
Hi,
Let us take a look into the code and discuss your issue. My or my college will come back will come back to you soon.

**Comment by casey-juul** [comment]
Thanks. I will say that it seems to be much more reliable in giving a disconnect error than Version2.

If you decide to implement a retry, it would be useful to add that as a DFUState or an additional delegate method so we can display the appropriate UI. In Version2, I had to parse one of the error messages which was less than ideal :)

**Comment by larsblumberg** [comment]
@casey-pax I implemented a workaround for this issue, I discussed it here: #71 

```
let dfuInitiator = DFUServiceInitiator(...)
...
var dfuController = dfuInitiator.start()
...
public func dfuError(_ error: DFUError, didOccurWithMessage message: String) {
    if error == .deviceDisconnected {
        dfuController.abort()
        dfuController = dfuInitiator.start()
    }
}
```

**Comment by mostafaberg** [comment]
This issue has seemed to be resolved in 3.0.4, will close and please reopen a new issue if problem persists
